### PR TITLE
PIMS-1816: Exemption Field On Add Project

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## 🎯 Summary

  
[PIMS-1816 ](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-1816)

* Adds an exemption note field towards the bottom of the add project page.
* Will only be enabled if the exemption task has also been ticked.

Note: The note that I believe should be the correct one here is mapped to a different status from the exemption task. Submitted exemption vs Approved for exemption. Anyone have any thoughts on whether this is acceptable or not?

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: react-app/src/components/projects/AddProject.tsx [hunk 1]
```diff
@@ -40,32 +41,40 @@ const AddProject = () => {
...
+    [lookupData],
+  );
+  const salesCostType = useMemo(
+    () => lookupData?.MonetaryTypes?.find((a) => a.Name === 'SalesCost'),
+    [lookupData],
+  );
+  const programCostType = useMemo(
+    () => lookupData?.MonetaryTypes?.find((a) => a.Name === 'ProgramCost'),
+    [lookupData],
+  );
```
Comment (dbarkowsky) [inline react-app/src/components/projects/AddProject.tsx:61]: Similar comment as in other PR regarding finding items using name. Changing name could cause issues.

File: react-app/src/components/projects/AddProject.tsx [hunk 2]
```diff
@@ -221,6 +230,14 @@ const AddProject = () => {
...
             </Grid>
           ))}
+          <Grid item xs={12}>
+            <TextFormField
+              disabled={watch.find((a) => a.TaskId === exemptionTask.Id)?.IsCompleted != true}
+              fullWidth
+              name={'ExemptionNote'}
+              label={'Exemption rationale note'}
+            />
+          </Grid>
```
Comment (dbarkowsky) [inline react-app/src/components/projects/AddProject.tsx:240]: In old PIMS, this was a required field when there was a request for exemption. 
Also suggest making it a multiline input and hiding when not required.

```suggestion
          {watch.find((a) => a.TaskId === exemptionTask.Id)?.IsCompleted === true ? (
            <Grid item xs={12}>
              <TextFormField
                fullWidth
                name={'ExemptionNote'}
                label={'Exemption rationale note'}
                required
                multiline
                minRows={2}
                rules={{
                  required: 'Rationale required when requesting exemption.',
                }}
              />
            </Grid>
          ) : (
            <></>
          )}
```

**Review by dbarkowsky — CHANGES_REQUESTED** [review]
Some suggestions about how the field is presented. Works well otherwise.

File: react-app/src/components/projects/AddProject.tsx (hunk 1 above)
Comment (GrahamS-Quartech) [reply react-app/src/components/projects/AddProject.tsx:61]: I think I'll just handle this pattern in all the spots its used in another ticket. Will make an enum for it as I believe the id's should remain consistent enough between migrations as long as we don't screw up the sequence somehow.

File: react-app/src/components/projects/AddProject.tsx (hunk 2 above)
Comment (GrahamS-Quartech) [reply react-app/src/components/projects/AddProject.tsx]: I think some web design purists may argue that hiding fields this way is bad practice but I personally think it's alright here. Implemented these changes.

File: react-app/src/components/projects/AddProject.tsx (hunk 1 above)
Comment (GrahamS-Quartech) [reply react-app/src/components/projects/AddProject.tsx]: Replaced these with the relevant enums.

**Review by dbarkowsky — APPROVED** [review]

**Pull request merged**